dbt/macros/format_additional_select_columns.sql

```diff
@@ -0,0 +1,55 @@
+-- Format a list of strings or objects representing columns that should be
+-- selected in a test, such that the columns can be templated into a SELECT
+-- statement.
+--
+-- If an element of the list is a string, the column it represents will
+-- be selected as-is. If instead the element is a dictionary, it can contain
+-- the following key-value pairs:
+--
+-- * `column` (required string): The name of the column to select
+-- * `alias` (optional string): The alias to use for the column
+-- (defaults to `column`)
+-- * `agg_func` (optional string): An aggregation function to use to
+-- select the column (defaults to no aggregation)
```

Comment:
An additional subtlety that is left out of this docstring: If not specifying an `agg_func`, the `alias` attribute is actually not necessary, meaning there's no real reason to use the dict structure unless you need aggregation. Nonetheless, it's supported here for the sake of maximum flexibility.

As an example, these two configurations will produce identical results:

```yaml
additional_select_columns:
  - foo as bar
  - baz as qux
```

```yaml
additional_select_columns:
  - column: foo
    alias: bar
  - column: baz
    alias: qux
```

It's necessary to support `alias` in the dict structure in order to allow for aggregation with a configurable name; we could consider failing if a dict structure is used but `agg_func` is not present, which would have the benefit of enforcing one standard approach, but I figure making the interface forgiving is more important than enforcing a specific standard. This also leaves open the possibility of supporting additional transformations on top of `agg_func` in the future while ensuring that such a change would be easy to make.